discovery/test_server.go

```diff
@@ -0,0 +1,55 @@
+package discovery
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+)
+
+// TestServer represents a Consul server we can run our tests against. Depends
+// on a local `consul` binary installed into our environ's PATH.
+type TestServer struct {
+	cmd      *exec.Cmd
+	HTTPAddr string
+}
+
+// NewTestServer constructs a new TestServer by including the httpPort as well.
+func NewTestServer(httpPort int) (*TestServer, error) {
+	path, err := exec.LookPath("consul")
+	if err != nil || path == "" {
+		return nil, fmt.Errorf("consul not found on $PATH - download and install " +
+			"consul or skip this test")
+	}
+
+	args := []string{"agent", "-dev"}
```

Comment:
You're not configuring the port. Add `"-http-port", strconv.Itoa(httpPort)` to `args`

Reply:
Doh!